engine/src/main/java/org/hibernate/validator/internal/xml/ValidationBootstrapParameters.java

```diff
@@ -204,6 +215,22 @@ private void setParameterNameProvider(String parameterNameProviderFqcn, ClassLoa
 		}
 	}
 
+	private void setClockProvider(String clockProviderFqcn, ClassLoader externalClassLoader) {
+		if ( clockProviderFqcn != null ) {
+			try {
+				@SuppressWarnings("unchecked")
+				Class<? extends ClockProvider> clazz = (Class<? extends ClockProvider>) run(
+						LoadClass.action( clockProviderFqcn, externalClassLoader )
+				);
+				clockProvider = run( NewInstance.action( clazz, "clock provider class" ) );
+				log.usingClockProvider( clazz );
+			}
+			catch (ValidationException e) {
+				throw log.getUnableToInstantiateParameterNameProviderClassException( clockProviderFqcn, e );
```

Comment:
The usage of getUnableToInstantiateException and the other similar one is in fact very specific. I changed it to getUnableToInstantiateClockProviderClassException.